icu4c/source/tools/toolutil/pkg_genc.cpp

```diff
@@ -294,14 +304,25 @@ writeAssemblyCode(
         exit(U_FILE_ACCESS_ERROR);
     }
 
+    const char* newSuffix = nullptr;
+
+    if (uprv_strcmp(assemblyHeader[assemblyHeaderIndex].name, "masm") == 0) {
+        newSuffix = ".masm";
```

Comment:
@rp9-next this changes the suffix for MASM. Does this look right to you?

Reply:
The change in suffix is needed for Meson, since it is possible there to use MASM and NASM simultaneously. It is detailed here: https://mesonbuild.com/Release-notes-for-0-64-0.html#new-languages-nasm-and-masm